### PR TITLE
[sysdig-deploy] Add admission controller

### PR DESCRIPTION
**Pull request by ironashram**
## What this PR does / why we need it:

- Add admission-controller as dependency of sysdig-deploy
- Add namespace field to all resources in admission-controller
- Fix README.md header navigation in sysdig-deploy
- Bump admission-controller version
- Bump sysdig-deploy version

## Checklist

[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]

- [x] Title of the PR starts with chart name (e.g. [mychartname])
- [x] Chart Version bumped for the respective charts
- [x] Changelog updated for the charts with version updates.
- [x] Variables are documented in the README.md (or README.tpl in some charts)
- [x] Check GithubAction checks (like lint) to avoid merge-check stoppers

Check Contribution guidelines in README.md for more insight.

### CODE REVIEW

**Review by lilx1ao — APPROVED** [review]

File: charts/sysdig-deploy/README.md
Comment (aroberts87) [inline charts/sysdig-deploy/README.md]: nit: there is an extra space between 'Secure' and the ending period. Nearly every other description here doesn't end with a period, so we could very well just get rid of it.

File: charts/admission-controller/templates/_helpers.tpl
Comment (aroberts87) [inline charts/admission-controller/templates/_helpers.tpl]: super minor nit: all other parameters/values have a space on either side of the `|`, but you're missing one after `.Values.clusterName` here

**Review by aroberts87 — APPROVED** [review]
Two super minor nits. I'm approving since I'd be okay if you merged without fixing.

**Comment by ironashram** [comment]
> Two super minor nits. I'm approving since I'd be okay if you merged without fixing.

thanks, fixed both

File: charts/admission-controller/templates/scanner/deployment.yaml
Comment (dark-vex) [inline charts/admission-controller/templates/scanner/deployment.yaml]: @ironashram I believe you should change also this line otherwise the proxy will get set only when using `admissionController.scanner.httpProxy` and gets ignored when using the global settings
cc: @lilx1ao @aroberts87

Comment (ironashram) [reply charts/admission-controller/templates/scanner/deployment.yaml]: indeed, great catch thank you

Comment (lilx1ao) [reply charts/admission-controller/templates/scanner/deployment.yaml]: nice!

Comment (dark-vex) [reply charts/admission-controller/templates/scanner/deployment.yaml]: Thanks for fixing it Michele! 

One more question, I see you have also set the namespace variable for multiple objects, like: secrets, service, role, PodSecurityPolicy, etc.. 

What's the purpose of that? You would like to give the possibility to deploy the admission controller on a different namespace while using `sysdig-deploy`? I'm asking because by default without specifying it, helm will take care of patching the manifest files and add the namespace specified with `helm install` command

File: charts/admission-controller/templates/scanner/psp.yaml
Comment (dark-vex) [inline charts/admission-controller/templates/scanner/psp.yaml:2]: Don't hate me but could you please add the k8s version check to this PSP policy? Like what we did for the agent https://github.com/sysdiglabs/charts/blob/master/charts/agent/templates/psp.yaml#L1
Otherwise the deploy on Kubernetes v1.25 will fail due to the deprecation of PSP policies

**Review by dark-vex — CHANGES_REQUESTED** [review]

**Review by yashgiri — APPROVED** [review]
Thank you for the changes 👍

File: charts/admission-controller/templates/scanner/psp.yaml
Comment (ironashram) [reply charts/admission-controller/templates/scanner/psp.yaml:2]: done!

File: charts/admission-controller/templates/scanner/deployment.yaml
Comment (ironashram) [reply charts/admission-controller/templates/scanner/deployment.yaml]: the idea was to give that option yes, but also to try and align with the other charts which are already among the dependencies

File: charts/admission-controller/values.yaml
Comment (dark-vex) [inline charts/admission-controller/values.yaml:23]: I did a test by setting a different namespace for the admission controller and in case the namespace doesn't already exist the whole helm installation fails:
```
Error: INSTALLATION FAILED: namespaces "test" not found
```
this is the command used:
```
 ➜ helm install sysdig-agent -f values-test.yaml charts/sysdig-deploy -n sysdig-agent --create-namespace
```

values-test.yaml
```
global:
  clusterConfig:
    name: "test-cluster"
  sysdig:
    accessKeySecret: "sysdig-agent"
    region: "us1"
  image:
    registry: quay.io
  proxy:
    httpProxy: http://squid.mydomain.tld:3128
    httpsProxy: http://squid.mydomain.tld:3128
    noProxy: 100.96.0.0/10

admissionController:
  enabled: true
  namespace: "test"
  sysdig:
    secureAPIToken: test

nodeAnalyzer:
  nodeAnalyzer:
    benchmarkRunner:
      deploy: false
      resources:
        limits:
          cpu: 250m
    runtimeScanner:
      deploy: true
      eveConnector:
        deploy: true
      resources:
        limits:
          cpu: 300m
      settings:
        eveEnabled: true
  secure:
    vulnerabilityManagement:
      newEngineOnly: true
```

I don't like too much the idea of giving the possibility to specify a different namespace while using sysdig-deploy chart but seems to be fine for the other maintainers so we can keep it :) 
Just please ensure to add a note on the `readme.md` file that for the specified namespace, it must exist prior to apply/install the chart

**Review by dark-vex — COMMENTED** [review]
Added a comment for the namespace flag, aside that, the PR LGTM

File: charts/admission-controller/values.yaml
Comment (lilx1ao) [reply charts/admission-controller/values.yaml:23]: good point. this is a problem. may need to discuss with the product team before committing to this.

**Review by dark-vex — APPROVED** [review]

File: charts/admission-controller/values.yaml
Comment (ironashram) [reply charts/admission-controller/values.yaml:23]: i have added a note for the namespace flag, it should be quite clear now

**Pull request merged**
